aw-server/src/config.rs

```diff
@@ -116,3 +118,21 @@ pub fn create_config(testing: bool) -> AWConfig {
 
     aw_config
 }
+
+/// Retrieves the device ID, if none exists it generates one (using UUID v4)
+pub fn get_device_id() -> String {
+    // TODO: Use seperate device_id for testing mode
+    // TODO: Cache to avoid retrieving on every /info call
```

Comment:
Why not handle it the same way as the config, read it once and then pass it around to those who need it? Makes more sense from a Rust standpoint and is less error prone.
Could possibly even be a part of the config structure if you wanted.